app/main.py

```diff
@@ -2,114 +2,325 @@
-import json
 import logging
-
+from app.models import *
+from math import ceil
+import json
 
 api = Blueprint('api')
 DEFAULT_TOPIC = "audio-recordings"
+ITEMS_PER_PAGE = 20
 
-client = MongoClient('mongodb://localhost:27017/')
 
-my_db = client.linguadb
+responseError = {"message": "Invalid Payload."}
+responseListObjects = {"page": 1, "results": [],
+                       "total_results": 0, "total_pages": 0}
 
 
 @api.route("/")
 def index(request):
     return response.json({"message": "welcome to Lingua API"})
 
 
+@api.route("/languages", methods=["POST"])
+def saveLanguages(request):
+    if request.method == "POST":
+        postdata = request.json
+        if "language" not in postdata:
+            return response.json(responseError, status=400)
+        if "code" not in postdata:
+            return response.json(responseError, status=400)
+        if "type" not in postdata:
+            return response.json(responseError, status=400)
+        try:
+            new_lang = Languages(
+                language=postdata["language"],
+                code=postdata["code"],
+                type=postdata["type"])
+            if "default" in postdata:
+                new_lang.default = postdata["default"]
+            new_lang.save()
+            return response.json({"message": "Added One Item"}, status=201)
+        except Exception as err:
+            responseError["message"] = str(err)
+            return response.json(responseError, status=400)
+
+
 @api.route("/languages")
-def get_languages(request):
+def getLanguages(request):
     """
-    Return all languages
-    Endpoint example: /languages
-    if query parameters are given, the endpoint will return that specific object
-    Example:  /languages?code=FRA
-    Parameter Name: code
+    Return all Languages objects
+    Parameters in Request body
+    {
+        "page" : (int)  [Optional, default is 1]
+    }
     """
-    queryfilter = {}
-    if len(request.args) != 0:
-        if "code" not in request.args:
-            return response.json({"message": "invalid query parameters "})
-        else:
-            queryfilter["code"] = request.args["code"][0]
+    try:
+        languages = Languages.objects()
+        args = request.json
+        if args:
+            if len(args) > 2:
+                raise Exception("Expecting less than 3 arguments")
+            if "language" in args:
+                languages = languages.filter(language=args["language"])
+            if "type" in args and "language" in args:
+                responseError["message"] = "Cannot filter with both arguments `language` and `type`"
+                return response.json(responseError, status=400)
+            if "type" in args:
+                languages = languages.filter(type=args["type"])
+            if "page" in args and int(args["page"]) > 1:
+                languages = languages.skip(int(args["page"])*ITEMS_PER_PAGE)
+                responseListObjects["page"] = args["page"]
```

Comment:
Please can you explain this line of code, I don't understand what it does.

Reply:
My understanding is that you are taking the json and querying value by value. Is it possible to translate the Json request into a python object to make stuff easier to understand?